Source/WebKit/UIProcess/Network/NetworkProcessProxyCocoa.mm

```diff
@@ -150,6 +154,10 @@
     if ([webViewUIDelegate respondsToSelector:@selector(_hostSceneBundleIdentifierForWebView:)])
         bundleIdentifier = [webViewUIDelegate _hostSceneBundleIdentifierForWebView:webView.get()];
 
+#if PLATFORM(VISION)
+    page->uiClient().willPresentModalUI(*page);
+#endif
```

Comment:
The platform guard is unnecessary, given `ENABLE_APPLE_PAY_REMOTE_UI_USES_SCENE` is only true for visionOS